### PR TITLE
Support https links for Source/Sinks

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation
Currently we support submitting sources/sinks with an archive thats a weblink. This support is restricted to http links only. This pr adds support for https links as well.
### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/Utils.java [hunk 1]
```diff
@@ -331,7 +332,8 @@ public static NarClassLoader extractNarClassLoader(Path archivePath, String pkgU
                     throw new IllegalArgumentException(
                             "Corrupt User PackageFile " + pkgUrl + " with error " + e.getMessage());
                 }
-            } else if (pkgUrl.startsWith("http")) {
+            } else if (pkgUrl.startsWith(org.apache.pulsar.common.functions.Utils.HTTP)
```
Comment (jerrypeng) [inline pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/Utils.java]: Why is this necessary?  "startsWith" checks for the prefix "http" which would include "https" no?

Comment (jerrypeng) [reply pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/Utils.java]: I have tested https with source and sinks work

**Comment by srkukarni** [comment]
@jerrypeng you are right!
